storage-access.bs

```diff
@@ -162,7 +162,26 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 
     ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). In practice, this might involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
 
-1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global|'s [=environment/has storage access=].
+1. Let |topLevelSite| be the result of [=obtain a site|obtaining a site=] from the [=top-level origin=] of |doc|'s [=relevant settings object=].
+1. Let |embeddedSite| be the result of [=obtain a site|obtaining a site=] from |doc|'s [=Document/origin=].
+1. Run the following steps [=in parallel=]:
+    1. Let |whether the user agent explicitly allows unpartitioned cookie access| be an algorithm that, given a [=tuple=] |tuple| consisting of [=sites=], runs the following steps. This algorithm returns "none", "allow" or "disallow".
+        
+        Note: A user agent's settings might explicitly allow or disallow unpartitioned cookie access through per-site allow-lists, the user changing global browser settings, or similar custom overrides.
+        
+        1. If the user agent does not have explicit settings for unpartitioned cookie access for |tuple|, return "none".
+        1. If the user agent's settings explicitly allow unpartitioned cookie access for |tuple|, return "allow".
+        1. If the user agent's settings explicitly disallow unpartitioned cookie access for |tuple|, return "disallow".
```

Comment:
I guess "disallow" is unused here now, but we might use it if we re-use this algorithm for rSA later?

Reply:
That's right. "disallow" is not checked here explicitly, but we can add

> 1. If |explicitSetting| is "disallow", [=/resolve=] |p| with false.

if that's clearer/preferred.

Reply:
Hmm not sure, it seems unnecessary to add there. It's probably fine like this? Maybe @annevk has a preference...

Reply:
I'm somewhat surprised this has explicit "allow". I thought that was always subject to `requestStorageAccess()` for child frames?

Reply:
There are user agent settings that disable 3PC blocking (e.g. in Safari IIRC) or allow-list specific sites (e.g. in Firefox and Chrome), how else would we represent those?